### PR TITLE
feat: Created an issuer verification of an EBSI issued credential

### PR DESCRIPTION
**Pull request by zoemaas**

### CODE REVIEW

File: packages/oid4vci-holder/src/agent/OID4VCIHolder.ts [hunk 1]
```diff
@@ -189,6 +190,30 @@ export function signCallback(
   }
 }
 
+export async function verifyEBSICredentialIssuer(args: VerifyEBSICredentialIssuerArgs): Promise<VerifyEBSICredentialIssuerResult | undefined> {
+  const { wrappedVc } = args
+
+  const issuer = wrappedVc.decoded?.iss ?? (typeof wrappedVc.decoded?.vc?.issuer === 'string' ? wrappedVc.decoded?.vc?.issuer : wrappedVc.decoded?.vc?.issuer?.existingInstanceId)
+
+  if (!issuer) {
```
Comment (nklomp) [inline packages/oid4vci-holder/src/agent/OID4VCIHolder.ts:198]: Why are you using an approach that we have never used?

Return a promise reject with an error so you do not have to list undefined as a result type. Or resolve undefined if that is an okay outcome. Not a reject with undefined. That makes no sense.

Comment (maikel-maas) [reply packages/oid4vci-holder/src/agent/OID4VCIHolder.ts:198]: Refactored error handling

File: packages/oid4vci-holder/src/agent/OID4VCIHolder.ts [hunk 2]
```diff
@@ -189,6 +190,30 @@ export function signCallback(
...
+  const url = `https://api-conformance.ebsi.eu/trusted-issuers-registry/v4/issuers/${issuer}`;
+  const response = await fetch(url)
+  if (response.status !== 200) {
+    return Promise.reject(undefined)
+  }
+
+  const payload = await response.json()
+
+  if (!payload.attributes.some((a: Attribute) => ['RootTAO', 'TAO', 'TI'].includes(a.issuerType))) {
+   return Promise.reject(undefined)
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply packages/oid4vci-holder/src/agent/OID4VCIHolder.ts]: Refactored error handling and moved issuer type to a parameter

File: packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts [hunk 3]
```diff
@@ -139,6 +139,15 @@ export const verifyCredentialToAccept = async (args: VerifyCredentialToAcceptArg
     }
   }
 
+  if (onVerifyEBSICredentialIssuer) {
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts]: Moved it to the if clause to make sure the code will only execute for an EBSI issuer

**Review by nklomp — APPROVED** [review]

**Pull request merged**